excavator/utils.py

```diff
@@ -64,8 +64,13 @@ def env_list(name, separator=',', required=False, default=empty):
     done by splitting the string value from the environment on a given
     separator.
     """
-    value = get_env_value(name, required=required, default=default)
+    if required and default is not empty:
+        raise ValueError("Using `default` with `required=True` is invalid")
+
+    value = get_env_value(name, required=required, default=empty)
     if value is empty:
+        if default is not empty:
+            return default
         return []
     # wrapped in list to force evaluation in python 3
     return list(filter(bool, [v.strip() for v in value.split(separator)]))
```

Comment:
(Did these comments bottom-to-top, but following up on the comment on the `env_int` version:

```python
def env_list(name, separator=',', required=False, default=empty):
    """
    Return a list of items derived from an environmental variable.  This is
    done by splitting the string value from the environment on a given
    separator.
    """
    value = get_env_value(name, required=required, default=default)
    if value is empty:
        return []
    # wrapped in list to force evaluation in python 3
    try:
        return [
            v.strip()
            for v
            in value.split(separator)
            if v.strip()
        ]
    except AttributeError:
        # We don't have a string. We will trust that we have a list-ish default,
        # then, and return it:
        return value
```

)